gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/EngineImpl.java

```diff
@@ -64,6 +67,7 @@ public EngineImpl(final IRegistry registry, final IPluginRegistry pluginRegistry
         
         policyFactory.setPluginRegistry(pluginRegistry);
         metrics.setComponentRegistry(componentRegistry);
+        initializeMetric(metrics);
```

Comment:
Could you clarify what you mean by this? For instance, you'd like initialize called on each factory, or on each of the factory's entries (e.g. every policy, every component, etc)